### ISSUE TITLE
Composite Primary Key Group for use with Foreign Key

### ISSUE THREAD

**Issue by mrkma** [Raizlabs/DBFlow#142]
Is this possible in DBFlow? I have two columns defined as foreign keys, and would like the primary key to be a composite key of the two.

I do understand an alternative would be to have a auto-incrementing  primary key with the two foreign keys in a 'unique group', but as reported here https://github.com/Raizlabs/DBFlow/issues/141 the unique groups aren't working properly.

**Comment by agrosner** [comment]
Ah interesting. Ill see whats happening with #141

**Comment by mrkma** [comment]
This isn't actually solved with the solution of #141.

You can't apply unique groups to foreign key fields properly, because the actual column definition is in the `ForeignKeyReference`, which doesn't have support for unique groups, and if you put a unique group in the `@Column` annotation it fails because it looks at the field's name, which isn't actually written into the table's definition.

I'll supply some code later for recreation.

**Comment by dmapr** [comment]
@mrkma, did you find a solution or a workaround? It doesn't seem to be working in the 2.2.1

```
@Table(databaseName = DatabaseInfo.NAME, uniqueColumnGroups = @UniqueGroup(groupNumber = 1, uniqueConflict = ConflictAction.FAIL))
public class Field extends BaseModel {
    @Column
    @PrimaryKey(autoincrement = true)
    long id;

    @Column
    @Unique(unique = false, uniqueGroups = 1)
    String name;

    @Column
    String type;

    @Column
    boolean hide;

    @Column
    @Unique(unique = false, uniqueGroups = 1)
    @ForeignKey(references = {@ForeignKeyReference(columnName = "category_id", columnType = Long.class, foreignColumnName = "id")}, onDelete = ForeignKeyAction.CASCADE, saveForeignKeyModel = false)
    ForeignKeyContainer<Category> categoryContainer;

    public void associateCategory(Category category) {
        categoryContainer = new ForeignKeyContainer<Category>(Category.class);
        categoryContainer.setModel(category);
    }
}
```

produces the following error:

>android.database.sqlite.SQLiteException: table Field has no column named categoryContainer (code 1): , while compiling: CREATE TABLE IF NOT EXISTS `Field`(`id` INTEGER PRIMARY KEY AUTOINCREMENT, `name` TEXT, `type` TEXT, `hide` INTEGER,  `category_id` INTEGER, UNIQUE (`name`, `categoryContainer`) ON CONFLICT FAIL, FOREIGN KEY(`category_id`) REFERENCES `Category` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE );
 
Or perhaps I'm doing something wrong?
